pkg/commands/deploy_actions.go

```diff
@@ -0,0 +1,1012 @@
+		logger.Error("config migration failed: %w", err)
+	}
+	if configsMigratedCount > 0 {
+		logger.Info("configs migrated: %d", configsMigratedCount)
+	}
+
+	// Migrate contexts
+	contextsMigratedCount, err := contexts.MigrateContexts(logger)
+	if err != nil {
+		logger.Error("context migrations failed: %w", err)
+	}
+	if contextsMigratedCount > 0 {
+		logger.Info("contexts migrated: %d", contextsMigratedCount)
+	}
+
+	// Load config for selected context
+	config, err := common.LoadConfigWithContextConfig(contextName)
+	if err != nil {
+		return err
+	}
+	if contextName == "" {
+		contextName = config.Config.Project.Context
+	}
+
+	// Check for context
+	yamlPath, rootNode, contextNode, err := common.LoadContext(contextName)
+	if err != nil {
+		return fmt.Errorf("context loading failed: %w", err)
+	}
+
+	// Fetch EigenLayer addresses using Zeus if requested
+	if useZeus {
+		err = common.UpdateContextWithZeusAddresses(cCtx.Context, logger, contextNode, contextName)
+		if err != nil {
+			logger.Warn("Failed to fetch addresses from Zeus: %v", err)
+			logger.Info("Continuing with addresses from config...")
+		} else {
+			logger.Info("Successfully updated context with addresses from Zeus")
+
+			// Write yaml back to project directory
+			if err := common.WriteYAML(yamlPath, rootNode); err != nil {
+				return fmt.Errorf("failed to save updated context: %v", err)
+			}
+		}
+	}
+
+	// Get fork_urls for the provided network (default to sepolia testnet)
+	l1ForkUrl, err := common.GetForkUrlDefault(contextName, config, common.L1)
+	if err != nil {
+		return fmt.Errorf("L1 fork URL error %w", err)
+	}
+	l2ForkUrl, err := common.GetForkUrlDefault(contextName, config, common.L2)
+	if err != nil {
+		return fmt.Errorf("L2 fork URL error: %w", err)
+	}
+
+	// Get chains node
+	chainsNode := common.GetChildByKey(contextNode, "chains")
+	if chainsNode == nil {
+		return fmt.Errorf("missing 'chains' key in context")
+	}
+
+	// Update RPC URLs for L1 chain
+	l1ChainNode := common.GetChildByKey(chainsNode, common.L1)
+	if l1ChainNode != nil {
+		l1RpcUrlNode := common.GetChildByKey(l1ChainNode, "rpc_url")
+		if l1RpcUrlNode != nil {
+			l1RpcUrlNode.Value = l1ForkUrl
+		}
+	}
+
+	// Update RPC URLs for L2 chain
+	l2ChainNode := common.GetChildByKey(chainsNode, common.L2)
+	if l2ChainNode != nil {
+		l2RpcUrlNode := common.GetChildByKey(l2ChainNode, "rpc_url")
+		if l2RpcUrlNode != nil {
+			l2RpcUrlNode.Value = l2ForkUrl
+		}
+	}
+
+	// Write yaml back to project directory
+	if err := common.WriteYAML(yamlPath, rootNode); err != nil {
+		return err
+	}
+
+	// Deploy the contracts after starting devnet unless skipped
+	if err := DeployL1ContractsAction(cCtx); err != nil {
+		return fmt.Errorf("deploy-contracts failed: %w", err)
+	}
+
+	// Sleep for 1 second to make sure new context values have been written
+	time.Sleep(1 * time.Second)
+
+	// Register AVS with EigenLayer
+	logger.Title("Registering AVS with EigenLayer...")
+	if !cCtx.Bool("skip-setup") {
```

Comment:
it would be nice if this could be skipped by a flag in the context. for tee compute, we're doing all this setup in a contract call on behalf of the developer and they shouldn't need to worry about passing `skip-setup` to get this command to run successfully

Reply:
Awesome, that makes sense! I will make the change 🙏